CHANGELOG.md

```diff
@@ -3,7 +3,64 @@
 All notable changes to this project will be documented in this file. This
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.51.0
+
+This release contains improvements to monitoring and an assortment of fixes and improvements.
+
+### Monitoring
+
+- Surface unauthorized response count from OPA API authz handler ([#3378](https://github.com/open-policy-agent/opa/issues/3378)) authored by @ashutosh-narkar
+- plugins: Surface more decision log errors via status API ([#5637](https://github.com/open-policy-agent/opa/issues/5637)) authored by @ashutosh-narkar
+- plugins/rest: Include truncated HTTP response in logs ([#2961](https://github.com/open-policy-agent/opa/issues/2961)) authored by @ashutosh-narkar reported by @gshively11
+
+### Topdown and Rego
+
+- wasm: Add native support for `object.union_n` built-in function (authored by @Azanul)
+
+### Fixes
+
+- ast/compile: Guard recursive module equality check. ([#5756](https://github.com/open-policy-agent/opa/issues/5756)) authored by @philipaconrad
+- ast: Relaxing strict-mode check for unused args in else-branching functions. ([#5758](https://github.com/open-policy-agent/opa/issues/5758)) authored by @johanfylling reported by @ethanjli
+- ast: Properly set the reported location of unused variables in strict-mode errors. ([#5662](https://github.com/open-policy-agent/opa/issues/5662)) authored by @boranx
+- fmt: report wrong arity for built-in functions. ([#5646](https://github.com/open-policy-agent/opa/issues/5646)) authored by @Trolloldem
+- topdown: http.send(): Ensuring intra-query caching consistency. ([#5736](https://github.com/open-policy-agent/opa/issues/5736)) authored by @johanfylling
+- Use Normalized policy paths as compiler module keys and store ids. (authored by @ashutosh-narkar)
+
+### Docs, Website, and Ecosystem
+
+- Fix typo in documentation (authored by @eternaltyro)
+- Update TLS authentication docs (authored by @charlieegan3)
+- Update _index.md (authored by @Ronnie-personal)
```

Comment:
Let's mention that the link for the windows sha sums was explicitly mentioned or something like that.